src/sentry/auth/providers/saml2.py

```diff
@@ -46,9 +95,97 @@ def get_provider(organization_slug):
         return None
 
 
+class OptionsForm(forms.Form):
+    options_identifier = forms.ChoiceField(label="Identifier attribute", choices=IDENTIFIER_CHOICES, help_text=_('Select what field will be used in order to match the user account.'))
+    options_jit = forms.BooleanField(label="Just-in-time Provisioning", required=False, help_text=_('Auto-provisioning. If user not exists, Sentry will create a new user account with the data provided by the IdP.'))
+    options_slo = forms.BooleanField(label="Single Logout Service", required=False, help_text=_('Enable/disable Single Log Out. SLO  is a complex functionality, the most common SLO implementation is based on front-channel (redirections), sometimes if the SLO workflow fails a user can be blocked in an unhandled view. If the admin does not controls the set of apps involved in the SLO process maybe is better to disable this functionality (can carry more problems than benefits).'))
+
+    def clean(self):
+        super(OptionsForm, self).clean()
+
+        options_identifier = self.data.get('options_identifier', None)
+        if (options_identifier is None or options_identifier == 'email') and not self.data.get('attribute_mapping_email', None):
+            self._errors["options_identifier"] = [_("Email selected as identifier but its mapping not provided")]
+            del self.cleaned_data["options_identifier"]
+        elif (options_identifier == 'username') and not self.data.get('attribute_mapping_username', None):
+            self._errors["options_identifier"] = [_("Username selected as identifier but its mapping not provided")]
+            del self.cleaned_data["options_identifier"]
+
+        if self.data.get('options_jit', None) and not (self.data.get('attribute_mapping_email', None) and self.data.get('attribute_mapping_username', None) and self.data.get('attribute_mapping_displayname', None)):
+            self._errors["options_jit"] = [_("JIT enabled but required attribute mapping not provided")]
+            del self.cleaned_data["options_jit"]
+
+        if self.data.get('options_slo', None) and not self.data.get('idp_slo_url', None):
+            self._errors["options_slo"] = [_("SLO enabled but Single Logout Service URL not provided")]
+            del self.cleaned_data["options_slo"]
+
+        return self.cleaned_data
+
+
+class AttributeMappingForm(forms.Form):
+    attribute_mapping_email = forms.CharField(label='Email', required=False)
+    attribute_mapping_username = forms.CharField(label='Username', required=False)
+    attribute_mapping_displayname = forms.CharField(label='Display Name', required=False)
+
+
+class SAML2ConfigureView(ConfigureView):
+    saml_form_cls = None
+    advanced_form_cls = None
+
+    def dispatch(self, request, organization, auth_provider):
+        if self.saml_form_cls is None or self.advanced_form_cls is None:
+            raise NotImplementedError('Custom forms may be defined by the extended class')
+
+        if request.POST:
```

Comment:
I assume you're trying to detect if the method was POST, but this is not the right way. You should check against `request.method == 'POST'`